colony/commands/bp.py

```diff
@@ -38,34 +46,57 @@ def do_validate(self):
         name = self.args.get("<name>")
         branch = self.args.get("--branch")
         commit = self.args.get("--commit")
+        remote = self.args.get("--remote")
 
         if commit and branch is None:
             raise DocoptExit("Since commit is specified, branch is required")
 
+        temp_working_branch = ""
+        repo = None
         if branch:
             working_branch = branch
         else:
             # Try to detect branch from current git-enabled folder
             logger.debug("Branch hasn't been specified. Trying to identify branch from current working directory")
             try:
-                working_branch = get_blueprint_working_branch(os.getcwd(), blueprint_name=name)
+                repo = BlueprintRepo(os.getcwd())
+                # todo get flag for stash mode
+                working_branch = get_blueprint_working_branch(repo, blueprint_name=name)
                 self.message(f"Automatically detected current working branch: {working_branch}")
+
             except BadBlueprintRepo as e:
                 working_branch = None
                 logger.warning(
                     f"No branch has been specified and it could not be identified from the working directory; "
                     f"reason: {e}. A branch of the Blueprints Repository attached to Colony Space will be used"
                 )
 
+            if not remote and not repo.is_current_branch_synced():
```

Comment:
repo and working_branch here could None or undefined

Reply:
I think this should solve it:
if not remote and repo and not repo.is_current_branch_synced():
but might have other implications

Reply:
double checking I think I meant:
if not remote and working_branch and not repo.is_current_branch_synced():